src/com/wsscode/pathom/core.cljc

```diff
@@ -177,16 +186,23 @@
        :else
        (parser env query)))))
 
-(defn join-seq [{::keys [entity-key] :as env} coll]
+(defn join-seq
+  "Runs the current subquery against the items of the given collection."
+  [{::keys [entity-key] :as env} coll]
   (mapv #(join (-> env
+                 ;; FIXME: Perhaps ::entity? Isn't entity-key a map? Do you really want a map as your key?
```

Comment:
The reason to use `::entity-key` is `::entity` by default, the reason is to make it customizable, the case for it would be if in some special condition you need to parse from a different key on the env that's not `::entity`, then you can set the `::entity-key` to something else. I'm not 100% sure about this feature and that's a reason I didn't document it much yet.